### PR TITLE
Feat/nightlies stats

### PR DESCRIPTION
**Pull request by mzedel**

### CODE REVIEW

File: ui/pages/nightlies.js [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+import { request, gql } from 'graphql-request';
+
+import { Grid, IconButton, Stack, Tooltip, Typography } from '@mui/material';
+import { Circle } from '@mui/icons-material';
+
+const buildStatusColorMap = {
+  FAILED: 'error',
+  RUNNING: 'warning',
+  SUCCESS: 'success',
+  default: 'warning.dark' // WTF is going on colour!
```
Comment (oleorhagen) [inline ui/pages/nightlies.js:12]: I like it!

File: ui/pages/nightlies.js [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+};
+
+const buildStatusColor = status => buildStatusColorMap[status] || buildStatusColorMap.default;
+
+const BuildStatus = ({ nightlies }) => {
+  const openNightlyClick = item => window.open(`https://gitlab.com${item.path}`, '_newtab');
```
Comment (oleorhagen) [inline ui/pages/nightlies.js:18]: Nice

File: ui/pages/nightlies.js [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+    <>
+      <Typography variant="h4">Nightlies</Typography>
+      <Stack>
+        {Object.values(nightlies)
+          .reverse()
+          .map(monthlyNightlies => {
+            const date = new Date(monthlyNightlies[0].startedAt);
+            return (
+              <React.Fragment key={date.getMonth()}>
+                <Typography variant="h6">{date.toLocaleString('default', { month: 'long' })}</Typography>
```
Comment (oleorhagen) [inline ui/pages/nightlies.js:30]: Nice, I'm gonna adopt something similar for the DB `timestamps`

File: ui/pages/nightlies.js [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+
+const buildStatusColor = status => buildStatusColorMap[status] || buildStatusColorMap.default;
+
+const BuildStatus = ({ nightlies }) => {
+  const openNightlyClick = item => window.open(`https://gitlab.com${item.path}`, '_newtab');
+
+  return (
+    <>
+      <Typography variant="h4">Nightlies</Typography>
+      <Stack>
```
Comment (oleorhagen) [inline ui/pages/nightlies.js:23]: Could we have the entire `Stack` stand out from the rest of the page a little bit? Like giving it a border or something? :thinking:

File: ui/pages/nightlies.js [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+                    <Stack key={item.path} alignItems="center">
+                      <Tooltip
+                        arrow
+                        title={
+                          <>
+                            {new Date(item.startedAt).toLocaleString()}
+                            {Object.entries(item.testReportSummary.total).map(([name, value]) => (
+                              <Stack direction="row" justifyContent="space-between" key={name}>
+                                <b>{name}</b>
+                                <div>{Math.ceil(value)}</div>
```
Comment (oleorhagen) [inline ui/pages/nightlies.js:42]: Q: Why? would we ever get fractions here?

File: ui/pages/index.js [hunk 6]
```diff
@@ -4,17 +4,19 @@ import { useRouter } from 'next/router';
...
   { title: 'Projects', location: '/projects', component: ProjectsView },
   { title: 'Build Status', location: '/build-status', component: BuildStatus },
+  { title: 'Nigthlies', location: '/nightlies', component: Nightlies },
   { title: 'Security Status', location: '/security-status', component: SecurityStatus }
 ];
 
 const Index = () => {
   const router = useRouter();
-  const tabIndex = paths.findIndex(({ location }) => location === router.pathname);
+  const tabIndex = paths.findIndex(({ location }) => location.startsWith(router.pathname));
```
Comment (oleorhagen) [inline ui/pages/index.js:19]: Why?

File: docker-compose.yml [hunk 7]
```diff
@@ -70,6 +70,12 @@ services:
             proxy_buffer_size 16k;
             proxy_pass http://ui:80;
           }
+          location /_next/webpack-hmr {
```
Comment (oleorhagen) [inline docker-compose.yml:73]: hah, I too found this in the end!

**Review by oleorhagen — APPROVED** [review]
Sleak

File: ui/pages/nightlies.js (hunk 5 above)
Comment (mzedel) [reply ui/pages/nightlies.js:42]: the duration comes in a fraction 👌

File: ui/pages/index.js (hunk 6 above)
Comment (mzedel) [reply ui/pages/index.js:19]: because it would redirect on partial matches otherwise 😕

**Pull request merged**